applications/crossbar/src/modules/cb_token_auth.erl

```diff
@@ -42,30 +49,66 @@ init() ->
     _ = crossbar_bindings:bind(<<"*.allowed_methods.token_auth">>, ?MODULE, 'allowed_methods'),
     _ = crossbar_bindings:bind(<<"*.resource_exists.token_auth">>, ?MODULE, 'resource_exists'),
     _ = crossbar_bindings:bind(<<"*.validate.token_auth">>, ?MODULE, 'validate'),
+    _ = crossbar_bindings:bind(<<"*.execute.post.token_auth">>, ?MODULE, 'post'),
     _ = crossbar_bindings:bind(<<"*.execute.delete.token_auth">>, ?MODULE, 'delete'),
     ok.
 
 -spec allowed_methods() -> http_methods().
+-spec allowed_methods(path_token()) -> http_methods().
+-spec allowed_methods(path_token(), path_token()) -> http_methods().
 allowed_methods() -> [?HTTP_DELETE, ?HTTP_GET].
 
+allowed_methods(?REFRESH_PATH_TOKEN) -> [?HTTP_POST];
+allowed_methods(?REFRESH_TOKENS_PATH_TOKEN) -> [?HTTP_DELETE].
+
+allowed_methods(?REFRESH_TOKENS_PATH_TOKEN, _RefreshToken) -> [?HTTP_DELETE].
+
 -spec resource_exists() -> 'true'.
+-spec resource_exists(path_token()) -> 'true'.
+-spec resource_exists(path_token(), path_token()) -> 'true'.
 resource_exists() -> 'true'.
 
+resource_exists(?REFRESH_PATH_TOKEN) -> 'true';
+resource_exists(?REFRESH_TOKENS_PATH_TOKEN) -> 'true'.
+
+resource_exists(?REFRESH_TOKENS_PATH_TOKEN, _RefreshToken) -> 'true'.
+
 -spec validate(cb_context:context()) -> cb_context:context().
+-spec validate(cb_context:context(), path_token()) -> cb_context:context().
+-spec validate(cb_context:context(), path_token(), path_token()) -> cb_context:context().
 validate(Context) ->
     cb_context:put_reqid(Context),
-    validate(Context, cb_context:req_verb(Context)).
+    validate_1(Context, cb_context:req_verb(Context)).
```

Comment:
This naming goes against how most crossbar modules name the specific validators. I would name this `validate_tokens/2`, and `validate_3` should probably be `validate_refresh_token`.